Modules/_cursesmodule.c

```diff
@@ -4732,9 +4733,28 @@ PyInit__curses(void)
         return NULL;
     ModDict = d; /* For PyCurses_InitScr to use later */
 
+    void **PyCurses_API = PyMem_Calloc(PyCurses_API_pointers, sizeof(void *));
+    if (PyCurses_API == NULL) {
+        PyErr_NoMemory();
+        return NULL;
+    }
+    /* Initialize the C API pointer array */
+    PyCurses_API[0] = (void *)Py_NewRef(&PyCursesWindow_Type);
+    PyCurses_API[1] = (void *)func_PyCursesSetupTermCalled;
+    PyCurses_API[2] = (void *)func_PyCursesInitialised;
+    PyCurses_API[3] = (void *)func_PyCursesInitialisedColor;
+
     /* Add a capsule for the C API */
-    c_api_object = PyCapsule_New(PyCurses_API, PyCurses_CAPSULE_NAME, NULL);
-    PyDict_SetItemString(d, "_C_API", c_api_object);
+    c_api_object = PyCapsule_New(PyCurses_API, PyCurses_CAPSULE_NAME,
+                                 curses_destructor);
+    if (c_api_object == NULL) {
+        PyMem_Free(PyCurses_API);
```

Comment:
Hum, there is no a refleak here. Does it work to call curses_destructor() here?

Reply:
Oh, thanks, updated.
Weird. `./python Lib/test/regrtest.py -ucurses test_curses -vv -R 3:3` can not catch this refleak.

Reply:
> Does it work to call curses_destructor() here?

No, it's can not work because capsule object haven't created

Reply:
>  Weird. ./python Lib/test/regrtest.py -ucurses test_curses -vv -R 3:3 can not catch this refleak.

test_curses doesn't test memory allocation failures. See _testcapi.set_nomemory() for that.

Reply:
Ah, and _curses doesn't use multiphase init API yet.

Reply:
Got it.